test/unit/lookup_value_test.rb

```diff
@@ -217,4 +217,22 @@ def setup
       refute_valid @value
     end
   end
+
+  context "when key type is puppetclass lookup and  value is empty" do
+    def setup
+      @key = FactoryGirl.create(:puppetclass_lookup_key, :as_smart_class_param,
+                                :override => true, :key_type => 'string',
+                                :default_value => 'whatever', :puppetclass => puppetclasses(:one), :use_puppet_default => true)
+      @value = LookupValue.new(:value => "", :match => "hostgroup=Common", :lookup_key_id => @key.id, :use_puppet_default => true)
```

Comment:
This passes using `FactoryGirl.build_stubbed(:lookup_value, :value => '', etc...`, should be [slightly faster]( https://robots.thoughtbot.com/use-factory-girls-build-stubbed-for-a-faster-test).